lib/ghomfp.gi

```diff
@@ -818,41 +855,61 @@ InstallMethod(IsomorphismFpGroupByGeneratorsNC,"subgroups of fp group",
   [IsSubgroupFpGroup,IsList and IsMultiplicativeElementWithInverseCollection,
    IsString],0,
 function(u,gens,nam)
-local aug,w,p,pres,f,fam;
+local aug,w,p,pres,f,fam,G;
   if HasIsWholeFamily(u) and IsWholeFamily(u) and
     IsIdenticalObj(gens,GeneratorsOfGroup(u)) then
       return IdentityMapping(u);
   fi;
   # get an augmented coset table from the group. It must be compatible with
   # `gens', so we must always use MTC.
-  if HasGeneratorsOfGroup(u) and IsIdenticalObj(GeneratorsOfGroup(u),gens) then
-    aug:=AugmentedCosetTableMtcInWholeGroup(u);
-  else
-    w:=FamilyObj(u)!.wholeGroup;
-    aug:=AugmentedCosetTableMtc(w,SubgroupNC(w,gens),2,"%");
-    # do not store the generators for the subgroup (the user could do this
-    # himself if he wanted), the danger of consequential errors due to a
-    # wrong <gens> list is too high.
-  fi;
-
-  # force computation of words for the secondary generators
-  SecondaryGeneratorWordsAugmentedCosetTable(aug);
-
-  # create a tietze object to reduce the presentation a bit
-  if not IsBound(aug.subgroupRelators) then
-    aug.subgroupRelators := RewriteSubgroupRelators( aug, aug.groupRelators);
+  
+  # use new MTC
+  w:=FamilyObj(u)!.wholeGroup;
+  aug:=NEWTC_CosetEnumerator(FreeGeneratorsOfFpGroup(w),
+	RelatorsOfFpGroup(w),
+	List(GeneratorsOfGroup(u),UnderlyingElement),
+	true);
+
+  pres:=NEWTC_PresentationMTC(aug,1,"%");
+  if Length(GeneratorsOfPresentation(pres))>Length(gens) then
+    aug:=NEWTC_CosetEnumerator(FreeGeneratorsOfFpGroup(w),
+	  RelatorsOfFpGroup(w),
+	  List(GeneratorsOfGroup(u),UnderlyingElement),
+	  true,false);
+
+    pres:=NEWTC_PresentationMTC(aug,0,"%");
   fi;
-  aug:=CopiedAugmentedCosetTable(aug);
-  pres := PresentationAugmentedCosetTable( aug,nam,0,true );
-  TzOptions(pres).printLevel:=InfoLevel(InfoFpGroup);
-  DecodeTree(pres);
+  Assert(0,Length(GeneratorsOfPresentation(pres))=Length(gens));
+
+  # old code
```

Comment:
This "old code" is a bit cryptic. Could we perhaps have a slightly longer explanation? Something like: "In 2016, the MTC code was rewritten from scratch as the old code was determined to be buggy, and nobody knew how it worked, so it was also hard to maintain. We leave parts of the old code in for now, because... ?"
(I'll leave it to somebody else to fill in the dots, as I am not sure myself why we leave it in :-).
Note that such a comment would only have to be written in one place; other relevant places could just refer to it ("For details, see the comment before function XYZ" or so).